src/Illuminate/Validation/Rules/DatabaseRule.php

```diff
@@ -43,10 +45,31 @@ trait DatabaseRule
      */
     public function __construct($table, $column = 'NULL')
     {
-        $this->table = $table;
+        $this->table = $this->resolveTableName($table);
         $this->column = $column;
     }
 
+    /**
+     * Resolves the name of the table from the given string.
+     *
+     * @param  string  $table
+     * @return string
+     */
+    public function resolveTableName($table)
+    {
+        if (! Str::contains($table, '\\') || ! class_exists($table)) {
+            return $table;
+        }
+
+        $model = new $table;
+
+        if (! $model instanceof Model) {
+            return $table;
+        }
+
+        return $model->getTable();
+    }
```

Comment:
Feel this could have been cleaner by checking the opposite, i.e. checking if the argument is a `Model` instance rather than not; checking the string is a class and exists rather than not.

```php
public function resolveTableName($table)
{
    if (is_object($table) && $table instanceof Model) {
        return $table->getTable();
    }

    if (class_exists($table)) {
        return (new $table)->getTable();
    }

    return $table;
}
```